### ISSUE TITLE
Smarter integer literals

### ISSUE THREAD

**Issue by btnlq** [jared-hughes/polygolf#243]
Transform integer literals to:

1. Hexadecimal numbers: `0x123` (Lua, Nim, Python, Swift).
1. Base 36 numbers: `int('abc',36)` (Python), `36'abc` (Prolog).
1. Floating point numbers: `69e3`. Free for languages based on floats (Lua, JavaScript). Can be used in some expressions in some other languages: `a<1e6` (Python, C#).
1. Chars: `'Ϩ'` (C#), `0'Ϩ` (Prolog).
1. Language-specific literals: `2mb` (PowerShell).
1. Bit shift expressions: `3<<19`.
1. Power expressions: `8^7`.

Transform `(any_integer IntegerType)` to any integer from `IntegerType`:

```erlang
while ($a < (any_integer 832041..1346269)) { ... };  % Polygolf
```

```py
while a<1e6:...  # Python
```

```swift
while a<1<<20{...}  // Swift
```

**Comment by MichalMarsalek** [comment]
It's not currently possible for a type to appear as an argument to an op, but that's ok, we can let it be
```
(any_integer 832041  1346269)
```
instead. I like this proposed operator.

**Comment by MichalMarsalek** [comment]
> Bit shift expressions: 3<<19.
> Power expressions: 8^7.

This comes down to expressing a large integer `n` in the form `n = k * b^e + d`, where k,b,e,d are small. I have no idea how to do that except for bruteforce.

**Comment by MichalMarsalek** [comment]
I guess bruteforcing just `b` isn't the worst thing.

**Comment by MichalMarsalek** [comment]
I thought about the `any_int` feature and I'm not sure it's a good addition.
For your example,  you can do
```
while ($a < {10^6; / 2^20}) { ... };  % Polygolf
```
and achieve the same.
The `any_int` expression anyway requires the user to think about what values can be useful in order to define the range.

**Comment by btnlq** [comment]
> The `any_int` expression anyway requires the user to think about what values can be useful in order to define the range.

It doesn't. The user writes the range for which their solution works (832041..1346269 for Fibonacci, 9974..10007 for Prime Numbers, and so on) and Polygolf do the magic.

> This comes down to expressing a large integer `n` in the form `n = k * b^e + d`, where `k`,`b`,`e`,`d` are small.

> `{10^6; / 2^20;}`

I have only one gold Fibonacci solution that uses a value other than 2<sup>x</sup> or 10<sup>x</sup>. Do `decomposeInt` really need to explore other bases? User can do it themselves.

**Comment by MichalMarsalek** [comment]
> The user writes the range for which their solution works (832041..1346269 for Fibonacci, 9974..10007 for Prime Numbers, and so on) and Polygolf do the magic.

Hmm, yeah ok.

> By the way, does `decomposeInt` need to explore other bases? User can do it themselves.

Wdym?

**Comment by MichalMarsalek** [comment]
Closed by #244 .
